Review shellcheck default exclusions in 05-shellcheck.sh

I was wondering why a couple of the default exclusions are in place:
- [SC2102](https://github.com/koalaman/shellcheck/wiki/SC2102): does not exist 
- [SC2123](https://github.com/koalaman/shellcheck/wiki/SC2124): seems quite legitimate considering how much we use arrays for command arguments etc
- [SC2155](https://github.com/koalaman/shellcheck/wiki/SC2155): I think this one can be left in, however prior to submitting pull requests this should be checked via a manual run of shellcheck
- [SC2148](https://github.com/koalaman/shellcheck/wiki/SC2148): why don't we include shebangs in auter.yumdnfModule and auter.aptModule? I know they are not stand-alone scripts, but I can't see a downside to including it; it would also mean shellcheck/syntax highlighting work without any additional faffing. 
- [SC2044](https://github.com/koalaman/shellcheck/wiki/SC2044): not sure why this is being ignored - it seems simple to fix any loops that we have currently doing this.

--- a/auter
+++ b/auter
@@ -89,9 +89,7 @@ function read_config() {
     logit "WARNING: Using default config values."
   fi
 
-  # Convert space separated list from the config file to an indexed array
-  # https://github.com/koalaman/shellcheck/wiki/SC2086
-  IFS=' ' read -r -a PACKAGEMANAGEROPTIONS <<< "$PACKAGEMANAGEROPTIONS"
+  IFS=' ' read -r -a PACKAGEMANAGEROPTIONS <<< "${PACKAGEMANAGEROPTIONS[*]}"
 }
 
 function rotate_file {
@@ -207,7 +205,7 @@ function log_last_run() {
 default_signal_handling
 
 
-ARGS=$@
+ARGS="$*"
 OPTS=$(getopt -n "$0" -o h,v --long prep,apply,enable,disable,reboot,postreboot,version,help,stdout,no-wall,status,config: -- "$@")
 if [[ $? -ne 0 ]]; then
   echo "Try '$0 --help' for more information."

--- a/auter.aptModule
+++ b/auter.aptModule
@@ -1,3 +1,4 @@
+#!/bin/bash
 # This is a script that is intended to only be called by /usr/bin/auter and
 # contains linux package manager specific code for auter.
 
@@ -130,6 +131,7 @@ function apply_updates() {
         run_script "${SCRIPT}" "Post-Apply"
       done
 
+      # Excluding this check because the REBOOTCALL variable is used by the main auter script
       # shellcheck disable=SC2034
       [[ "${AUTOREBOOT}" == "yes" ]] && REBOOTCALL=1
 

--- a/auter.yumdnfModule
+++ b/auter.yumdnfModule
@@ -1,3 +1,4 @@
+#!/bin/bash
 # This is a script that is intended to only be called by /usr/bin/auter and
 # contains linux package manager specific code for auter.
 
@@ -170,6 +171,7 @@ function apply_updates() {
     logit "INFO: Updates complete (${PACKAGE_MANAGER} ${TRANSACTIONID}). You may need to reboot for some updates to take effect"
     log_last_run
 
+    # Excluding this check because the REBOOTCALL variable is used by the main auter script
     # shellcheck disable=SC2034
     [[ "${AUTOREBOOT}" == "yes" ]] && REBOOTCALL=1
 

--- a/contrib/conditionalReboot/99-conditionalReboot.sh
+++ b/contrib/conditionalReboot/99-conditionalReboot.sh
@@ -60,6 +60,7 @@ LIBCHECK=$(PATH=/usr/sbin:/usr/local/sbin:$PATH lsof | grep lib | grep DEL)
 # Check if any of the packages in the APPLIST were updated
 if [[ -n ${APPLIST} ]]; then
   for PACKAGEMATCH in $APPLIST; do
+    # Excluding SC2001 for readability purposes
     # shellcheck disable=SC2001
     PACKAGEMATCH=$(echo "$PACKAGEMATCH" | sed 's/*/.*/g')
     for PACKAGE in "${PACKAGESUPDATED[@]}"; do

--- a/tests/05-shellcheck.sh
+++ b/tests/05-shellcheck.sh
@@ -7,10 +7,9 @@ EXITCODE=0
 AUTERDIR="$(cd "$(dirname "$0")" ; cd ../ ; pwd -P )"
 CHANGEDFILES="$(find "${AUTERDIR}" -name CHANGEDFILES)"
 
-# Create a list of script files to be checked. These are files 
+# Create a list of script files to be checked.
 if [[ -f "${CHANGEDFILES}" ]]; then
   ALLFILES=$(find "${AUTERDIR}" -type f -not -path '*/\.*' | egrep "$(xargs <"${CHANGEDFILES}" | tr ' ' '|')")
-  
 else
   ALLFILES=$(find "${AUTERDIR}" -type f -not -path '*/\.*')
 fi
@@ -25,8 +24,10 @@ echo "${SCRIPTSTOTEST}"
 grep -q "auter.aptModule" "${CHANGEDFILES}" && SCRIPTSTOTEST+=" ${AUTERDIR}/auter.aptModule "
 grep -q "auter.yumdnfModule" "${CHANGEDFILES}" && SCRIPTSTOTEST+=" ${AUTERDIR}/auter.yumdnfModule "
 
-# Custom shellcheck exclusions
-SHELLCHECK_EXCLUSIONS=",SC2102,SC2124,SC2155,SC2148"
+# SC2102 is related to https://github.com/koalaman/shellcheck/issues/682. This
+# was previously removed from the online checker but still exists in the
+# standalone package.
+SHELLCHECK_EXCLUSIONS="SC2102"
 
 echo "===================================="
 for SCRIPT in ${SCRIPTSTOTEST}; do
@@ -35,11 +36,11 @@ for SCRIPT in ${SCRIPTSTOTEST}; do
   # This can be done bu adding "# shellcheck disable=SC2016" to the previous line in the script
   # ----------------------------------------------#
   # Excluding SC2016 due to line 11 of 10-rpmbuild.sh. Expansion is specifically blocked
-#  [[ "${SCRIPT}" =~ 10-rpmbuild.sh ]] && SHELLCHECK_EXCLUSIONS+=",SC2016"
+  # [[ "${SCRIPT}" =~ 10-rpmbuild.sh ]] && SHELLCHECK_EXCLUSIONS+=",SC2016"
   # ----------------------------------------------#
 
   SCRIPT=$(realpath "${SCRIPT}")
-  SHELLCHECK_OUTPUT=$(shellcheck -e SC2044"${SHELLCHECK_EXCLUSIONS}" "${SCRIPT}")
+  SHELLCHECK_OUTPUT=$(shellcheck -e "${SHELLCHECK_EXCLUSIONS}" "${SCRIPT}")
   if [[ $? -eq 0 ]]; then
     echo -e "$g[ OK ] $w Tested ${SCRIPT} for common coding mistakes"
   else

